From d05d9247662581dabdd0ddeaaf1e478233ca5167 Mon Sep 17 00:00:00 2001
From: Ben Greear <greearb@candelatech.com>
Date: Thu, 11 Jun 2020 16:20:49 -0700
Subject: [PATCH 10/15] intel-oot: Decrease firmware monitor debug area from
 4MB to 512k.

It appears the code can handle down to 2^11, so set max in
this case to 2^19 to save mem.

Signed-off-by: Ben Greear <greearb@candelatech.com>
---
 drivers/net/wireless/intel-oot/iwlwifi/pcie/trans.c | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/drivers/net/wireless/intel-oot/iwlwifi/pcie/trans.c b/drivers/net/wireless/intel-oot/iwlwifi/pcie/trans.c
index 442d9dd829f8c..33d4b7aaf3857 100644
--- a/drivers/net/wireless/intel-oot/iwlwifi/pcie/trans.c
+++ b/drivers/net/wireless/intel-oot/iwlwifi/pcie/trans.c
@@ -172,8 +172,8 @@ static void iwl_pcie_alloc_fw_monitor_block(struct iwl_trans *trans,
			continue;

		IWL_INFO(trans,
-			 "Allocated 0x%08x bytes for firmware monitor.\n",
-			 size);
+			 "Allocated 0x%08x bytes for firmware monitor, max_power: %d  min_power: %d  power: %d.\n",
+			 size, max_power, min_power, power);
		break;
	}

@@ -193,6 +193,8 @@ static void iwl_pcie_alloc_fw_monitor_block(struct iwl_trans *trans,

 void iwl_pcie_alloc_fw_monitor(struct iwl_trans *trans, u8 max_power)
 {
+	pr_err("iwl_pcie_alloc_fw_monitor, monitor size max_power requested: %d\n",
+	       max_power);
	if (!max_power) {
		/* default max_power is maximum */
		max_power = 26;
@@ -205,6 +207,13 @@ void iwl_pcie_alloc_fw_monitor(struct iwl_trans *trans, u8 max_power)
		 max_power))
		return;

+	/* Try to use less memory, there are other things in the system as well! */
+	if (max_power > 19) {
+		pr_err("iwl_pcie_alloc_fw_monitor, decreasing max-power from %d to 19 to save memory.\n",
+		       max_power);
+		max_power = 19;
+	}
+
	if (trans->dbg.fw_mon.size)
		return;

--
2.26.3
